File: backend/routes/plans.js
/**
 * Subscription Plans Routes
 */

const express = require('express');
const router = express.Router();
const { getCollection } = require('../services/database');

// Get all subscription plans
router.get('/', (req, res) => {
  const plans = getCollection('subscriptionPlans');
  res.json(plans);
});

// Get popular plan
router.get('/popular', (req, res) => {
  const plans = getCollection('subscriptionPlans');
  const popularPlan = plans.find(p => p.popular);
  if (popularPlan) {
    res.json(popularPlan);
  } else {
    res.status(404).json({ error: 'No popular plan found' });
  }
});

module.exports = router;

File: backend/routes/plans.py
# Subscription Plans Routes

from fastapi import APIRouter, HTTPException
from typing import List
from models.schemas import SubscriptionPlan
from services.database import get_collection

router = APIRouter(prefix="/subscription-plans", tags=["Subscription Plans"])


@router.get("", response_model=List[SubscriptionPlan])
async def get_subscription_plans():
    """Get all subscription plans"""
    return get_collection("subscriptionPlans")


@router.get("/popular")
async def get_popular_plan():
    """Get the most popular plan"""
    plans = get_collection("subscriptionPlans")
    for plan in plans:
        if plan.get("popular"):
            return plan
    raise HTTPException(status_code=404, detail="No popular plan found")
